packages/devtools/lib/src/main.dart

```diff
@@ -128,8 +130,75 @@ class PerfToolFramework extends Framework {
     showError(title, error);
   }
 
-  void initTestingModel() {
-    App.register(this);
+  Future<bool> isFlutterApp() async {
+    final EvalOnDartLibrary flutterLibrary = EvalOnDartLibrary(
+      flutterLibraryUriCandidates,
+      serviceManager.service,
+    );
+
+    try {
+      await flutterLibrary.libraryRef.catchError(
+        (_) => throw FlutterLibraryNotFound(),
+        test: (e) => e is LibraryNotFound,
+      );
+    } catch (e) {
+      if (e is FlutterLibraryNotFound) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  Future<bool> isProfileBuild() async {
+    try {
+      final Isolate isolate = await serviceManager.service
+          .getIsolate(serviceManager.isolateManager.selectedIsolate.id);
+      await serviceManager.service.evaluate(
+        serviceManager.isolateManager.selectedIsolate.id,
+        isolate.rootLib.id,
+        '1+1',
+      );
+      // If we reach this return statement, no error was thrown and this is not
+      // a profile build.
+      return false;
+    } catch (e) {
+      if (e is RPCError) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  Future<void> addScreens() async {
+    final _isFlutterApp = await isFlutterApp();
+    final _isProfileBuild = await isProfileBuild();
+
+    addScreen(InspectorScreen(
+      disabled: !_isFlutterApp || _isProfileBuild,
+      disabledTooltip: !_isFlutterApp
+          ? 'This section is disabled because you are not running a Flutter '
+              'application'
+          : 'This section is disabled because you are running a profile build '
+          'of your application',
+    ));
+    addScreen(TimelineScreen(
+      disabled: !_isFlutterApp,
+      disabledTooltip: 'This section is disabled because you are not running a '
+          'Flutter application',
+    ));
+    addScreen(MemoryScreen());
+    if (showPerformancePage) {
+      addScreen(PerformanceScreen());
+    }
+    addScreen(DebuggerScreen(
+      disabled: _isProfileBuild || tabDisabledByQuery('debugger'),
```

Comment:
we also want a disabled by query option that flutterWeb can use to warn that the debugger isn't yet ready in flutter_web
this one would have a different tooltip.

Reply:
disabled pages for flutter web by looking for the package:flutter_web lib.